python/lbann/modules/graph/dense/DenseGCNConv.py

```diff
@@ -0,0 +1,27 @@
+import lbann 
+from  lbann.modules import Module 
+from lbann.util import str_list
+import math 
+
+
+class DenseGCNConv(Module):
+    global_count = 0
+
+    def __init__(self, input_channels, output_channels, name=None):
+        super().__init__()
+        DenseGCNConv.global_count += 1
+
+        self.name = (name if name else 'Dense_GCN_{}'.format(DenseGCNConv.global_count))
+        
+        
+        bounds = math.sqrt(6.0 / (input_channels + output_channels))
+        self.weights = lbann.Weights(initializer = lbann.UniformInitializer(min=-bounds,max=bounds),
+                                    name=self.name+'_Weights')
+
+        self.W = lbann.WeightsLayer(dims = str_list([input_channels, output_channels]),
+                               name = self.name+'_layer',
+                               weights =self.weights)
+    def forward(self,X,A):
```

Comment:
Added in a few comments saying we assume adjacency matrix is normalized. I tried to think of a way to normalize the matrix on the fly and add a flag, but this seemed a better solution. 

It Could be good to add that capability if/when we write the custom layers.